mage_ai/authentication/providers/google.py

```diff
@@ -0,0 +1,75 @@
+import urllib.parse
+import uuid
+from typing import Dict
+
+import aiohttp
+
+from mage_ai.authentication.oauth.constants import OAUTH_PROVIDER_GOOGLE
+from mage_ai.authentication.providers.oauth import OauthProvider
+from mage_ai.authentication.providers.sso import SsoProvider
+from mage_ai.authentication.providers.utils import get_base_url
+from mage_ai.settings.sso import GOOGLE_CLIENT_ID, GOOGLE_CLIENT_SECRET
+
+
+class GoogleProvider(SsoProvider, OauthProvider):
+    provider = OAUTH_PROVIDER_GOOGLE
+
```

Comment:
I would add a `__init__` method and validate the settings so that we can provide good exception messages.